code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+		if(isliving(G.assailant))
+			if(attempt_insert(user, G.affecting))
 				qdel(G)
-				state = OPEN_FULL
-			update_icon(UPDATE_ICON_STATE)
-		else
-			return ..()
-	else if(is_type_in_typecache(W, can_be_washed))
-		if(is_type_in_typecache(W, does_not_fit))
-			to_chat(user, "<span class='warning'>This item does not fit.</span>")
-			return
-		if(istype(W, /obj/item/clothing/gloves/color/black/krav_maga/sec))
-			to_chat(user, "<span class='warning'>Washing these gloves would fry the electronics!</span>")
-			return
-		if(W.flags & NODROP)
-			to_chat(user, "<span class='warning'>[W] is stuck to your hand!</span>")
+		return
+	if(istype(W))
+		if(attempt_insert(user, W))
 			return
+	return ..()
+
+
+/*
+	* # attempt_insert
+	*
+	* First part in the insert chain for putting items into the laundry machine, does some sanity checks on both the user and atom
+	* moveable that's about to be shoved inside. Makes sure the user can actually perform this action and then calls the next proper
+	* proc in the insert chain depending if they're a mob or obj/item
+*/
+/obj/machinery/washing_machine/proc/attempt_insert(mob/user, atom/movable/atom_to_insert)
+	if(!anchored || (stat & BROKEN) || (stat & NOPOWER))
+		return FALSE
+	if(!door_open)
+		to_chat(user, "<span class='warning'>[src]'s door is closed!</span>")
+		return FALSE // simply not possible (or is it?)
+	if(!istype(atom_to_insert) || !istype(user))
+		return FALSE
+	if(user.stat != CONSCIOUS || !Adjacent(user) || !atom_to_insert.Adjacent(user))
+		return FALSE
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		to_chat(user, "<span class='notice'>You try to insert [atom_to_insert] into [src] but your hands are blocked!</span>")
+		return FALSE
+	if(isliving(atom_to_insert)) // not ismob() b/c we only want living creatures, ghosts could just float out of the washing machine whenever they wanted
+		var/mob/living/sacrifice = atom_to_insert
+		if(sacrifice.mob_size >= MOB_SIZE_HUMAN) // prevents some most mobs from being washed even if it would be really really funny
+			to_chat(user, "<span class='notice'>You try to insert [sacrifice] into [src] but [sacrifice.p_theyre()] too large to fit inside the tub!</span>")
+			return FALSE
+		if(sacrifice.buckled)
+			to_chat(user, "<span class='notice'>You try to insert [sacrifice] into [src] but [sacrifice.p_theyre()] still buckled to something!</span>")
+			return FALSE
+		return insert_mob_into_tub(user, sacrifice)
+	if(isitem(atom_to_insert)) // Restricting this to only item types prevents a lot of issues, you really only wanna shove items in here anyway...
+		var/obj/item/item_to_insert = atom_to_insert
+		if(item_to_insert.anchored)
+			return FALSE
+		if(item_to_insert.w_class >= WEIGHT_CLASS_HUGE)
+			to_chat(user, "<span class='notice'>You try to insert [item_to_insert] into [src] but [item_to_insert.p_them()] is too large for the tub!</span>")
+			return FALSE
+		return insert_item_into_tub(user, item_to_insert)
+	return FALSE // how did we get here?
```

Comment:
stack_trace here would probably be good

Reply:
I just removed the return, this is technically unreachable code unless some bitshift happens or something